sway/input/switch.c

```diff
@@ -0,0 +1,79 @@
+#include "sway/config.h"
+#include "sway/desktop/transaction.h"
+#include "sway/input/switch.h"
+#include <wlr/types/wlr_idle.h>
+#include "log.h"
+
+struct sway_switch *sway_switch_create(struct sway_seat *seat,
+        struct sway_seat_device *device) {
+    struct sway_switch *lid_switch =
+        calloc(1, sizeof(struct sway_switch));
+    if (!sway_assert(lid_switch, "could not allocate switch")) {
+        return NULL;
+    }
+    device->lid_switch = lid_switch;
+    lid_switch->seat_device = device;
+    wl_list_init(&lid_switch->switch_toggle.link);
+    sway_log(SWAY_DEBUG, "Allocated switch for device");
+
+    return lid_switch;
+}
+
+static void handle_switch_toggle(struct wl_listener *listener, void *data) {
+
+    struct sway_switch *sway_switch = wl_container_of(listener, sway_switch, switch_toggle);
+
+    struct sway_seat* seat = sway_switch->seat_device->sway_seat;
+    //struct wlr_seat *wlr_seat = seat->wlr_seat;
```

Comment:
Noted, most of the commented out sections are just because I wasn't sure if I needed them or not, and because I took a lot of this code from `keyboard.c`.